src/librbd/image/RemoveRequest.cc

```diff
@@ -187,8 +187,10 @@ void RemoveRequest<I>::handle_detach_child(int r) {
   if (r < 0) {
     lderr(m_cct) << "failed to detach child from parent: "
                  << cpp_strerror(r) << dendl;
-    send_close_image(r);
-    return;
+    if (r != -ENOENT) {
```

Comment:
I'd prefer to hide this in the actual `DetachChildRequest` state machine. For example, if this was failing in `DetachChildRequest<I>::clone_v2_child_detach` when it attempts to create the `IoCtx` for the child, or `DetachChildRequest<I>::handle_clone_v2_get_snapshot` when it attempts to read the snapshot, handle the `-ENOENT` there.